marahgh/assignment1/Q1.cpp

```diff
@@ -0,0 +1,171 @@
+
+
+#include <iostream>
+#include<vector>
+using namespace std;
+
+/*anagram words */
+//My assumption:the words alphabet is english letters.    
+
+
+//---------------------------------------------------------------------------------------------------------
+/*A function that determines if two strings are anagrams ,
+	able to handle both case sensitive and case insensitive anagrams,
+	according to value of caseSensetive flag variable */
+bool areAnagrams(string s1, string s2, bool caseSensetive){
+    const int LEN= 'z' -'a' +1;
+
+    if(s1.size()!= s2.size()){
+        return false;
+    }
+    vector<int> histogram(LEN*2,0);
+    for(char c: s1){
+        if((c <'a' || c> 'z') && (c <'A' || c> 'Z')){
+            return false;
+        }
+        if(!caseSensetive){
+            char lowerLetter=tolower(c);
+            histogram[lowerLetter - 'a']++;
+            continue;
+        }
+        if(c <='a' && c>='z'){
+            histogram[c - 'a']++;
+        }else{
+           histogram[c - 'A'+ LEN]++; 
+        }
+    }
+    for(char c: s2){

```

Comment:
Lines 37-52 are very similar to 22-36. Maybe this can be extracted into a function?
It also looks like the code could be a bit simpler, maybe?